### PR TITLE
Contact info card 

### PR DESCRIPTION
**Pull request by ahulab**
# Description

Add functionality for displaying contact info

# To test
- navigate to /contacts/ > contact info tab
- see two new cards in addition to existing view
- Profile card should have data from firebase

### CODE REVIEW

**Comment by evandana** [comment]
Overall, great first pass! 

### Functionality notes:

- I see a console log message: `proxyConsole.js:56 Warning: Unsupported style property padding-right. Did you mean paddingRight? Check the render method of 'ContactProfileCard'.`
- You can remove the junk between the tabs and "Profile"
- Please add the necessary nodes into FB DB so you can accommodate all data. Give a best guess at the node names based on the patterns already present.
- Any questions on how to address the Events section?

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# based on the airbnb styleguide https://github.com/airbnb/javascript/blob/master/.editorconfig
```
Comment (evandana) [inline .editorconfig]: Works pretty well for me with vscode. Let's get agreement from one other person on this though.

File: src/actions/index.js [hunk 2]
```diff
@@ -93,9 +97,22 @@ export function refreshConfig(config) {
     }
 }
 
-/** CONTACTS */
+/** CONTACT */
```
Comment (evandana) [inline src/actions/index.js]: I love the detail here... adding the code comment sections and differentiating `contacts` from `contact` and following the camelCase to SNAKE_CASE conversion.

File: src/components/controller/Contact/ContactProfileCard/index.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+// import { connect } from 'react-redux';
```
Comment (evandana) [inline src/components/controller/Contact/ContactProfileCard/index.js]: Why is this here, or why is it commented out?

File: src/components/view/Contact/ContactNotesCard/index.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
+
```
Comment (evandana) [inline src/components/view/Contact/ContactNotesCard/index.js]: Sometimes I've also just followed a file path pattern of something like ContactNotesCard.js under Contact instead of the additional nested folders. Both are fine, and no changes needed.

File: src/components/view/Contact/ContactNotesCard/index.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+      <div>
+        <Card initiallyExpanded={true} >
+          <CardHeader
+              title='Notes'
+              titleColor={this.props.titleColor}
+              actAsExpander={true}
+              showExpandableButton={true}
+          />
+          <CardText expandable={true}>
+            notes props here
```
Comment (evandana) [inline src/components/view/Contact/ContactNotesCard/index.js]: This text should come from the contact's info, through `props`, right?

File: src/components/view/Contact/ContactProfileCard/index.js [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+import {Card, CardActions, CardHeader, CardText} from 'material-ui/Card';
+
+class ContactProfileCard extends React.Component {
+
+  render() {
+    const { contact } = this.props;
+
+    const infoKeyStyle = {
+      color: '#90A4AE',
```
Comment (evandana) [inline src/components/view/Contact/ContactProfileCard/index.js]: Can you pull colors from muiThemable? See pattern in the PMC-2018 app if you need a reference. It involves some small tweaks to the controller for this as well, but should be worthwhile and would be a great learning experience (that we'll all have to follow). 

I'd recommend having ContactInfo set up the muiThemable and then pass the muiTheme to its sub-components.

File: src/components/view/Contact/ContactProfileCard/index.js [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+        <Card initiallyExpanded={true} >
+          <CardHeader
+              title='Profile'
+              titleColor={this.props.titleColor}
+              actAsExpander={true}
+              showExpandableButton={true}
+          />
+          <CardText expandable={true}>
+            <div className="row">
+              <div className="col-xs-6">
```
Comment (evandana) [inline src/components/view/Contact/ContactProfileCard/index.js]: Could you try experimenting with the col sizing across different page widths, just to test out smallest size vs desktop? It's not formally required, but I bet it will be a quick check and will handle more mobile screens better.

File: src/components/view/Contact/ContactProfileCard/index.js [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+import {Card, CardActions, CardHeader, CardText} from 'material-ui/Card';
+
+class ContactProfileCard extends React.Component {
+
+  render() {
+    const { contact } = this.props;
+
+    const infoKeyStyle = {
+      color: '#90A4AE',
+      'padding-right': '.5rem'
```
Comment (evandana) [inline src/components/view/Contact/ContactProfileCard/index.js]: Try `paddingRight`. JSX instead of CSS. ;)

File: src/components/view/Contact/ContactProfileCard/index.js [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+              </div>
+            </div>
+            <div className="row">
+              <div className="col-xs-6">
+                <span style={infoKeyStyle}>Age of first injection</span>
+                {contact.firstInjectionAge}
+              </div>
+              <div className="col-xs-6">
+                <span style={infoKeyStyle}>Hispanic</span>
+                {contact.hispanic ? 'True' : 'False'}
```
Comment (evandana) [inline src/components/view/Contact/ContactProfileCard/index.js]: Nice! Picking up the new patterns!

File: src/components/view/ContactInfo/index.js [hunk 10]
```diff
@@ -1,45 +1,30 @@
 import React, { Component } from 'react';
 
-import { getImageForEnv } from 'static/images/index'
+import { getImageForEnv } from 'static/images/index';
+import ContactProfileCard from 'components/view/Contact/ContactProfileCard';
+import ContactNotesCard from 'components/view/Contact/ContactNotesCard';
+// import EventsCard from '';
+import { cyan600 } from 'material-ui/styles/colors'
```
Comment (evandana) [inline src/components/view/ContactInfo/index.js]: See note above about muiThemeable.

File: src/components/view/ContactInfo/index.js [hunk 11]
```diff
@@ -1,45 +1,30 @@
...
-                                <td>{contact.birthCountry}</td>
-                                <td>{contact.hispanic}</td>
-                            </tr>
-                        ))}
-                    </tbody>
-                </table>
-                
+
+                <div>
+                    <ContactProfileCard titleColor={cyan600} contact={contact} />
```
Comment (evandana) [inline src/components/view/ContactInfo/index.js]: Pass in muiThemeable instead of titleColor to make it more maintainable.

File: src/reducers/index.js [hunk 12]
```diff
@@ -10,7 +11,8 @@ const reducers = {
     isLoggedIn,
     config,
     contacts,
+    contact,
```
Comment (evandana) [inline src/reducers/index.js]: It might seem annoying, but would you mind alphabetizing these lists (imports and this list)? There's a great VSCode plugin that makes it super quick.

File: src/sagas/contact.js [hunk 13]
```diff
@@ -0,0 +1,22 @@
...
+import {
+    GET_CONTACT,
+} from '../constants';
+
+import { updateCurrentContact } from 'actions';
+
+function* getContact({ uid }) {
+    const contactRef = `/contacts/${uid}`
+    window._FIREBASE_DB_.ref(contactRef)
+        .once('value', (snapshot) => {
```
Comment (evandana) [inline src/sagas/contact.js]: Excellent use of `once` instead of `on`!

File: src/sagas/contact.js [hunk 14]
```diff
@@ -0,0 +1,22 @@
...
+
+import {
+    GET_CONTACT,
+} from '../constants';
+
+import { updateCurrentContact } from 'actions';
+
+function* getContact({ uid }) {
+    const contactRef = `/contacts/${uid}`
+    window._FIREBASE_DB_.ref(contactRef)
```
Comment (evandana) [inline src/sagas/contact.js]: Thoughts on `window._FIREBASE_DB_.ref( '/contacts/' + ${uid} )`?

Or, keep it as is and ensure that `uid` exists first?

File: src/components/view/Contact/ContactNotesCard/index.js (hunk 5 above)
Comment (ahulab) [reply src/components/view/Contact/ContactNotesCard/index.js]: ultimately, yeah it should

File: src/sagas/contact.js (hunk 14 above)
Comment (ahulab) [reply src/sagas/contact.js]: how about
```javascript
window._FIREBASE_DB_.ref( `/contacts/${uid || ''}`)
```

or would we want to throw an error if there's no uid specified?

**Comment by evandana** [comment]
I've typically avoided using the interpolation ticks.  What about `window.
_FIREBASE_DB_.ref( '/contacts/' + (uid || '') )`?

File: src/components/controller/Contact/ContactProfileCard/index.js (hunk 3 above)
Comment (ahulab) [reply src/components/controller/Contact/ContactProfileCard/index.js]: removed. thought I had needed it

File: src/components/view/Contact/ContactProfileCard/index.js (hunk 7 above)
Comment (ahulab) [reply src/components/view/Contact/ContactProfileCard/index.js]: it looks good to me. on a phone it will be 2 columns

File: src/reducers/contact.js [hunk 15]
```diff
@@ -0,0 +1,21 @@
...
+} from '../constants'
+
+// TODO: should this be something other than an empty object?
+const defaultCurrentContactState = {};
+
+function contact(state = defaultCurrentContactState, action) {
+    const { contact } = action;
+    switch (action.type) {
+        case UPDATE_CURRENT_CONTACT:
+            return Object.assign({}, contact);
```
Comment (evandana) [inline src/reducers/contact.js]: Optional: 

```
{
   ...contact,
}
```
